### PR TITLE
Add variable info to xml generated in variables.js

### PR DESCRIPTION
**Pull request by marisaleung**

### CODE REVIEW

File: core/variables.js [hunk 1]
```diff
@@ -123,12 +120,15 @@ Blockly.Variables.flyoutCategory = function(workspace) {
...
 
-  if (variableNameList.length > 0) {
+  if (variableModelList.length > 0) {
+    var firstVariable = variableModelList[0];
     if (Blockly.Blocks['variables_set']) {
       var gap = Blockly.Blocks['math_change'] ? 8 : 24;
       var blockText = '<xml>' +
             '<block type="variables_set" gap="' + gap + '">' +
-            '<field name="VAR">' + variableNameList[0] + '</field>' +
+            '<field name="VAR" variableType="' + firstVariable.type +
```
Comment (rachel-fenichel) [inline core/variables.js]: Make a helper function to generate these field strings (multiple lines, repeated multiple times, and easy to accidentally break).

Pass in a VariableModel and return a string.

Comment (marisaleung) [inline core/variables.js]: Done.

File: core/variables.js [hunk 2]
```diff
@@ -280,3 +278,17 @@ Blockly.Variables.promptName = function(promptText, defaultText, callback) {
     callback(newVar);
   });
 };
+
+/**
+ * Generate XML string for variable field.
+ * @param {!Blockly.VariableModel} variableModel The variable model to generate
+ *     an XML string from.
+ * @return {string} The generated XML.
```
Comment (rachel-fenichel) [inline core/variables.js]: @private annotation and underscore on the name.

File: core/variables.js [hunk 3]
```diff
@@ -280,3 +278,17 @@ Blockly.Variables.promptName = function(promptText, defaultText, callback) {
...
+ */
+Blockly.Variables.generateVariableFieldXml = function(variableModel) {
+  var xmlString = '<field name="VAR" ' + 'variableType="' +
+      variableModel.type + '" id="' + variableModel.getId() + '">'+
+      variableModel.name +
+      '</field>';
+  return xmlString;
+}
```
Comment (rachel-fenichel) [inline core/variables.js]: Missing semicolon

File: core/variables.js (hunk 2 above)
Comment (marisaleung) [inline core/variables.js]: Done.

File: core/variables.js (hunk 3 above)
Comment (marisaleung) [inline core/variables.js]: Done.

**Pull request merged**